src/Features/TerrainHelper.cpp

```diff
@@ -0,0 +1,134 @@
+#include "TerrainHelper.h"
+
+#include "ShaderCache.h"
+#include "State.h"
+
+void TerrainHelper::DataLoaded()
+{
+	// Get the default landscape texture set for terrain helper
+	const auto defaultLandTextureSet = RE::TESForm::LookupByEditorID<RE::BGSTextureSet>("LandscapeDefault");
+	if (defaultLandTextureSet != nullptr) {
+		spdlog::info("LandscapeDefault EDID texture set found");
+		defaultLandTexture = defaultLandTextureSet;
+	} else {
+		spdlog::info("LandscapeDefault EDID texture set not found, using default");
+		const auto bgsDefaultLandTex = *REL::Relocation<RE::TESLandTexture**>(RELOCATION_ID(514783, 400936));
+		defaultLandTexture = bgsDefaultLandTex->textureSet;
+	}
+}
+
+bool TerrainHelper::TESObjectLAND_SetupMaterial(RE::TESObjectLAND* land)
+{
+	if (land->loadedData == nullptr || land->loadedData->mesh[0] == nullptr) {
```

Comment:
Is land ever nullptr? You're asking for a CTD.

Reply:
Updated